### PR TITLE
SALTO-3918 author information index

### PR DESCRIPTION
**Pull request by dantal4**
Added author information index that maps element ids to author information (changedBy & changedAt).
Added `getElementAuthorInformation` function to workspace API.

---

_Additional context for reviewer_

---
_Release Notes_: 
Workspace:
- Added author information index that maps element ids to author information (changedBy & changedAt)
- Added `getElementAuthorInformation` function to workspace API

---
_User Notifications_: 
None

### CODE REVIEW

**Review by alonstern — COMMENTED** [review]
LGTM 👍 
@ori-moisis Do you want to have a look too?

File: packages/workspace/src/workspace/workspace.ts [hunk 1]
```diff
@@ -1131,20 +1160,21 @@ export const loadWorkspace = async (
...
+      id,
+      envName,
+      async envState => await envState.referenceTargets.get(id.getFullName()) ?? []
+    ),
+    getElementIncomingReferences: async (id, envName = currentEnv()) => validateBaseIdAndRunOnEnvState(
+      id,
+      envName,
+      async envState => await envState.referenceSources.get(id.getFullName()) ?? []
+    ),
+    getElementAuthorInformation: async (id, envName = currentEnv()) => validateBaseIdAndRunOnEnvState(
```
[the remark this answers is not in the document]
Comment (dantal4) [reply packages/workspace/src/workspace/workspace.ts:1173]: The `get` call in the UI remote map implementation is debounced so it is actually a `getMany` call.

File: packages/workspace/src/workspace/index_utils.ts [hunk 2]
```diff
@@ -26,3 +30,70 @@ export const getAllElementsChanges = async (
...
+const getFieldChangesFromTypeChange = (change: Change<ObjectType>): Change<Field>[] => {
+  if (isAdditionOrRemovalChange(change)) {
+    return Object.values(getChangeData(change).fields).map(field => (
+      isAdditionChange(change)
+        ? toChange({ after: field })
+        : toChange({ before: field })
+    ))
+  }
+  const { before, after } = change.data
+  const allFieldNames = Object.keys({ ...before.fields, ...after.fields })
```
Comment (ori-moisis) [inline packages/workspace/src/workspace/index_utils.ts:43]: could crash if we have many fields?
maybe change this to
```suggestion
  const allFieldNames = new Set(Object.keys(before.fields).concat(Object.keys(after.fields)))
```

**Review by ori-moisis — APPROVED** [review]

File: packages/workspace/src/workspace/index_utils.ts (hunk 2 above)
Comment (dantal4) [reply packages/workspace/src/workspace/index_utils.ts:43]: From what I saw we got `RangeError: Maximum call stack size exceeded` only when we used spread in a function input (e.g `a.push(...b)`), but not inside an object/list creation.

**Pull request merged**